pydoctor/themes/base/search.js

```diff
@@ -0,0 +1,442 @@
+        // and also on the the original data present on all-documents.html for initialisation, 
+        // where the data is stored in the innerHTML. 
+        _type = value.querySelector(selector).innerHTML;
+      }
+
+      if(_type.endsWith(type)){
+        match = true;
+      }
+    })
+    
+    return match;
+  
+  }).length;  
+
+  if (nb_things==0){
+    input.style.display = "none";
+  }
+  else{
+    input.style.display = "block";
+  }
+}
+
+function initFilterDropdown(results_list_p){
+
+  let results_list = Array.prototype.slice.call(results_list_p);
+
+  document.getElementById("search-filter-show-all").style.display = "none";
+
+  _initSearchFilter(results_list, document.getElementById("search-filter-show-interfaces"), ["Interface"]);
+  _initSearchFilter(results_list, document.getElementById("search-filter-show-classes"), ["Class"]);
+  _initSearchFilter(results_list, document.getElementById("search-filter-show-functions"), ["Function"]);
+  _initSearchFilter(results_list, document.getElementById("search-filter-show-methods"), ["Method"]);
+  _initSearchFilter(results_list, document.getElementById("search-filter-show-modules"), ["Module", "Package"]);
+  _initSearchFilter(results_list, document.getElementById("search-filter-show-attributes"), ["Attribute"]);
+}
+
+///////////////////// SEARCH //////////////////////////
+
+function search(){
+
+  let _url = new URL(document.URL);
+  console.log(_url);
+
+  var _setLongSearchInfosTimeout = null;
+  var _query = null;
+
+  // Get the query terms
+  if (!_url.searchParams.has('search-query')){
+    setStatus('No search query provided.')
+    return ;
+  }
+
+  _query = _url.searchParams.get('search-query');
+
+  if (!_query.length>0){
+    setStatus('No search query provided.')
+    return ;
+  }
+
+  // Set the search box text to the query terms
+  document.getElementById('search-box').value = _query;
+
+  console.log("Your query is: "+ _query)
+
+  if (!window.Worker) {
+    setStatus("Cannot search: JavaScript Worker API is not supported in your browser. ");
+    return ;
+  }
+
+  let results_list = document.getElementById('search-results'); 
+
+  setStatus("Searching...")
+
+  // Setup the search worker   
+  let worker = new Worker('search-worker.js');
+
+  worker.postMessage({
+    query: _query,
+  });
+  
+  worker.onmessage = function (response) {
+
+    if (_setLongSearchInfosTimeout){
+      clearTimeout(_setLongSearchInfosTimeout)
+    }
+    setInfos('')
+    setWarning('')
+    
+    console.log("Message received from worker: ")
+    console.dir(response.data)
+
+    if (!response.data.results){
+      setErrorStatus();
+      throw("No data received from worker")
+    }
+
+    if (response.data.results.length == 0){
+      setStatus('No results matches "' + _query + '"');
+      return ;
+    }
+    else{
+      setStatus('Fetching documents...');
+    }
+
+    // Get result data
+    httpGet("all-documents.html", function(response2) {
```

Comment:
Would be best to start loading "all-documents.html" right at the beginning of the search process. Not only once we receive the response from the Worker (see `worker.onmessage = function (response)`). This would cut a couple of seconds per search, which is not negligible.